novawallet/Modules/Staking/NominationPools/BondMore/Setup/NominationPoolBondMorePresenter.swift

```diff
@@ -0,0 +1,182 @@
+import BigInt
+import SoraFoundation
+
+final class NominationPoolBondMorePresenter: NominationPoolBondMoreBasePresenter {
+    var wireframe: NominationPoolBondMoreWireframeProtocol? {
+        baseWireframe as? NominationPoolBondMoreWireframeProtocol
+    }
+
+    var interactor: NominationPoolBondMoreInteractorInputProtocol? {
+        baseInteractor as? NominationPoolBondMoreInteractorInputProtocol
+    }
+
+    private var inputResult: AmountInputResult?
+    private var transferrableBalance: Decimal?
+
+    init(
+        interactor: NominationPoolBondMoreInteractorInputProtocol,
+        wireframe: NominationPoolBondMoreWireframeProtocol,
+        chainAsset: ChainAsset,
+        hintsViewModelFactory: NominationPoolsBondMoreHintsFactoryProtocol,
+        balanceViewModelFactory: BalanceViewModelFactoryProtocol,
+        dataValidatorFactory: NominationPoolDataValidatorFactoryProtocol,
+        localizationManager: LocalizationManagerProtocol,
+        logger: LoggerProtocol
+
+    ) {
+        super.init(
+            interactor: interactor,
+            wireframe: wireframe,
+            chainAsset: chainAsset,
+            hintsViewModelFactory: hintsViewModelFactory,
+            balanceViewModelFactory: balanceViewModelFactory,
+            dataValidatorFactory: dataValidatorFactory,
+            localizationManager: localizationManager,
+            logger: logger
+        )
+    }
+
+    func provideTransferrableBalance() {
+        guard let balance = assetBalance?.transferable.decimal(precision: chainAsset.asset.precision) else {
+            view?.didReceiveTransferable(viewModel: nil)
+            return
+        }
+        let viewModel = balanceViewModelFactory.amountFromValue(balance).value(for: selectedLocale)
+        view?.didReceiveTransferable(viewModel: viewModel)
+    }
+
+    func provideAmountInputViewModel() {
+        guard let inputAmount = getInputAmount() else {
+            return
+        }
+        let viewModel = balanceViewModelFactory.createBalanceInputViewModel(
+            inputAmount
+        ).value(for: selectedLocale)
+
+        view?.didReceiveInput(viewModel: viewModel)
+    }
+
+    func provideAssetViewModel() {
+        let balance = transferrableBalance ?? 0
+        let inputAmount = getInputAmount() ?? 0
+        let viewModel = balanceViewModelFactory.createAssetBalanceViewModel(
+            inputAmount,
+            balance: balance,
+            priceData: price
+        ).value(for: selectedLocale)
+
+        view?.didReceiveAssetBalance(viewModel: viewModel)
+    }
+
+    override func getInputAmount() -> Decimal? {
+        guard let inputResult = inputResult else {
+            return nil
+        }
+        switch inputResult {
+        case .rate:
+            let fee = fee?.decimal(precision: chainAsset.asset.precision) ?? 0
+            let availableAmountDecimal = transferrableBalance ?? 0
+            return inputResult.absoluteValue(from: availableAmountDecimal - fee)
```

Comment:
We can just call this line without switch...case

Reply:
Why? We should subtract fee when only rate was chosen

Reply:
Just always pass the difference availableAmountDecimal - fee into the ```absoluteValue``` function